tests/api_fastapi/core_api/routes/public/test_task_instances.py

```diff
@@ -1497,6 +1497,75 @@ def test_should_respond_200_with_different_try_numbers(self, test_client, try_nu
             "dag_run_id": "TEST_DAG_RUN_ID",
         }
 
+    @pytest.mark.parametrize("try_number", [1, 2])
+    def test_should_respond_200_with_mapped_task_at_different_try_numbers(
+        self, test_client, try_number, session
+    ):
+        tis = self.create_task_instances(session, task_instances=[{"state": State.FAILED}])
+        print(tis)
+        old_ti = tis[0]
+        print(old_ti)
+        for idx in (1, 2):
+            ti = TaskInstance(task=old_ti.task, run_id=old_ti.run_id, map_index=idx)
+            ti.rendered_task_instance_fields = RTIF(ti, render_templates=False)
+            ti.try_number = 1
+            for attr in ["duration", "end_date", "pid", "start_date", "state", "queue", "note"]:
+                setattr(ti, attr, getattr(old_ti, attr))
+            session.add(ti)
+        session.commit()
+        tis = session.query(TaskInstance).all()
+        print(tis)
+        # Record the task instance history
+        from airflow.models.taskinstance import clear_task_instances
+
+        clear_task_instances(tis, session)
+        # Simulate the try_number increasing to new values in TI
+        for ti in tis:
+            if ti.map_index > 0:
+                ti.try_number += 1
+                ti.queue = "default_queue"
+                session.merge(ti)
+        session.commit()
+        tis = session.query(TaskInstance).all()
+        print(tis)
+        # in each loop, we should get the right mapped TI back
+        for map_index in (1, 2):
+            # Get the info from TIHistory: try_number 1, try_number 2 is TI table(latest)
+            print(
+                "/public/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances"
+                f"/print_the_context/{map_index}/tries/{try_number}",
+            )
+            response = test_client.get(
+                "/public/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances"
+                f"/print_the_context/{map_index}/tries/{try_number}",
```

Comment:
can you add a comment here that once Authentication is added, we need to update this to pass user details like we are doing in legacy